File: ATSBackend/ATSBackend.Service/Models/VagaModel.cs
using System;
using System.Collections.Generic;

namespace ATSBackend.Service.Models
{
    public class VagaModel
    {
        public int IdVaga { get; set; }
        public string Descricao { get; set; }
        public string Profissao { get; set; }
        public bool Encerrada { get; set; }
        public DateTime DataCriacao { get; set; }
        public string Local { get; set; }
        public int IdCandidato { get; set; }
        public virtual IEnumerable<CandidatoModel> Candidatos { get; set; }
    }
}

File: ATSFrontend/src/app/components/vaga/vaga.model.ts
export interface Vaga {
    idVaga?: number;
    descricao: string;
    profissao: string;
    dataCriacao: string;
    local: string;
    idCandidato: number;
    candidatos: Candidato[];
}

export interface Candidato {
    idCandidato?: number;
    nome: string;
    dataNascimento: string;
}